packages/loot-core/src/client/data-hooks/filters.ts

```diff
@@ -20,16 +20,19 @@ function toJS(rows) {
 }
 
 export function useFilters(): TransactionFilterEntity[] {
-  const filters = toJS(
-    useLiveQuery(() => q('transaction_filters').select('*'), []) || [],
+  const { data } = useQuery<TransactionFilterEntity>(
+    () => q('transaction_filters').select('*'),
+    [],
   );
 
   /** Sort filters by alphabetical order */
-  function sort(filters) {
-    return filters.sort((a, b) =>
-      a.name.trim().localeCompare(b.name.trim(), { ignorePunctuation: true }),
+  const sort = useCallback((filters: TransactionFilterEntity[]) => {
+    return [...filters].sort((a, b) =>
+      a.name
+        .trim()
+        .localeCompare(b.name.trim(), undefined, { ignorePunctuation: true }),
     );
-  }
+  }, []);
 
-  return useMemo(() => sort(filters), [filters]);
+  return useMemo(() => sort(toJS(data ? [...data] : [])), [data, sort]);
```

Comment:
Since the callback is just used once, can we just inline it instead and avoid a `useCallback`?

Reply:
Updated :)